pylops/signalprocessing/dwtnd.py

```diff
@@ -0,0 +1,142 @@
+__all__ = ["DWTND"]
+
+import logging
+from math import ceil, log
+
+import numpy as np
+
+from pylops import LinearOperator
+from pylops.basicoperators import Pad
+from pylops.utils import deps
+from pylops.utils.typing import DTypeLike, InputDimsLike, NDArray
+
+from .dwt import _adjointwavelet, _checkwavelet
+
+pywt_message = deps.pywt_import("the dwtnd module")
+
+if pywt_message is None:
+    import pywt
+
+logging.basicConfig(format="%(levelname)s: %(message)s", level=logging.WARNING)
+
+
+class DWTND(LinearOperator):
+    """N-dimensional Wavelet operator.
+
+    Apply ND-Wavelet transform along N ``axes`` of a
+    multi-dimensional array of size ``dims``.
+
+    Note that the Wavelet operator is an overload of the ``pywt``
+    implementation of the wavelet transform. Refer to
+    https://pywavelets.readthedocs.io for a detailed description of the
+    input parameters.
+
+    Defaults to a 3D wavelet transform along the last three dimensions
+    of the input array.
+
+    Parameters
+    ----------
+    dims : :obj:`tuple`
+        Number of samples for each dimension
+    axes : :obj:`int`, optional
+        .. versionadded:: 2.0.0
```

Comment:
Take `.. versionadded:: 2.0.0` as this operator is brand new

Reply:
Done.